spec.html

```diff
@@ -39015,57 +39030,11 @@ <h1>Promise.prototype.finally ( _onFinally_ )</h1>
         <p>When the `finally` method is called with argument _onFinally_, the following steps are taken:</p>
         <emu-alg>
           1. Let _promise_ be the *this* value.
-          1. If Type(_promise_) is not Object, throw a *TypeError* exception.
+          1. If IsPromise(_promise_) is *false*, throw a *TypeError* exception.
```

Comment:
This isn’t going to work; finally must work with thenables.

Reply:
Why is that? `Promise.prototype.then` does this check as well. Shouldn't `Promise.prototype.then` work with thenables as well?

Reply:
Because of subclassing, that check must be restricted to .then so that it can be overridden.